WORKSPACE.bazel

```diff
@@ -39,8 +25,97 @@ maybe(
     strip_prefix = "googletest-{version}".format(version = GOOGLETEST_VERSION),
 )
 
-# Fetch iceoryx_hoofs and iceoryx_platform
 
+# Load Rust rules
+# Use v0.26 to support bazel v6.2
+maybe(
+    name = "rules_rust",
+    repo_rule = http_archive,
+    sha256 = "9d04e658878d23f4b00163a72da3db03ddb451273eb347df7d7c50838d698f49",
+    urls = ["https://github.com/bazelbuild/rules_rust/releases/download/0.26.0/rules_rust-v0.26.0.tar.gz"],
+)
+
+load("@rules_rust//rust:repositories.bzl", "rules_rust_dependencies", "rust_register_toolchains")
+
+rules_rust_dependencies()
+rust_register_toolchains(
+    edition = "2021",
+    versions = [
+        "1.80.0"
+    ],
+)
+
+
+# Load prebuilt bindgen
+maybe(
+    name = "bindgen",
+    repo_rule = http_archive,
+    sha256 = "b7e2321ee8c617f14ccc5b9f39b3a804db173ee217e924ad93ed16af6bc62b1d",
+    strip_prefix = "bindgen-cli-x86_64-unknown-linux-gnu",
+    urls = ["https://github.com/rust-lang/rust-bindgen/releases/download/v0.69.5/bindgen-cli-x86_64-unknown-linux-gnu.tar.xz"],
+    build_file_content = """
+filegroup(
+    name = "bindgen-cli",
+    srcs = ["bindgen"],
+    visibility = ["//visibility:public"],
+)
+    """,
+)
+
+# Load prebuilt cbindgen
+maybe(
+    name = "cbindgen",
+    repo_rule = http_file,
+    sha256 = "521836d00863cb129283054e5090eb17563614e6328b7a1610e30949a05feaea",
+    urls = ["https://github.com/mozilla/cbindgen/releases/download/0.26.0/cbindgen"],
+    executable = True,
+)
+
+# Load external crates
+load("@rules_rust//crate_universe:repositories.bzl", "crate_universe_dependencies")
+
+crate_universe_dependencies()
+
+load("@rules_rust//crate_universe:defs.bzl", "crates_repository")
+
+maybe(
+    name = "crate_index",
+    repo_rule = crates_repository,
+    cargo_lockfile = "//:Cargo.lock",
+    lockfile = "//:Cargo.Bazel.lock",
+    manifests = [
+        "//:Cargo.toml",
+        "//:benchmarks/event/Cargo.toml",
+        "//:benchmarks/publish-subscribe/Cargo.toml",
+        "//:examples/Cargo.toml",
+        "//:iceoryx2/Cargo.toml",
+        "//:iceoryx2-bb/container/Cargo.toml",
+        "//:iceoryx2-bb/derive-macros/Cargo.toml",
+        "//:iceoryx2-bb/elementary/Cargo.toml",
+        "//:iceoryx2-bb/lock-free/Cargo.toml",
+        "//:iceoryx2-bb/log/Cargo.toml",
+        "//:iceoryx2-bb/memory/Cargo.toml",
+        "//:iceoryx2-bb/posix/Cargo.toml",
+        "//:iceoryx2-bb/system-types/Cargo.toml",
+        "//:iceoryx2-bb/testing/Cargo.toml",
+        "//:iceoryx2-bb/threadsafe/Cargo.toml",
+        "//:iceoryx2-bb/trait-tests/Cargo.toml",
+        "//:iceoryx2-cal/Cargo.toml",
+        "//:iceoryx2-cli/Cargo.toml",
+        "//:iceoryx2-ffi/ffi-macros/Cargo.toml",
+        "//:iceoryx2-ffi/ffi/Cargo.toml",
+        "//:iceoryx2-pal/concurrency-sync/Cargo.toml",
+        "//:iceoryx2-pal/configuration/Cargo.toml",
+        "//:iceoryx2-pal/posix/Cargo.toml",
```

Comment:
I understand the problem but I refuse to adjust the architecture since one build tool for a language binding cannot handle it.
But I would be open to merge multiple crates into one if this makes sense from an architecture point of view.

But weren't you the guy who promoted many small crates? When we ignore bazel for a minute, would you still promote this position?

Reply:
Yeah, small crates have some advantages since they are build in parallel. I'm not quite sure if this would also work if there is only one Cargo.toml with multiple libs, e.g. one in iceoryx2-bb but with a separate lib for memory, posix, etc.
Nevertheless, for cargo it doesn't really matter. The only downside is the number of crates to publish.